### PR TITLE
First stab at Hub section in spec

### PR DESCRIPTION
**Pull request by csuwildcat**
This is to be regarded as a strawman PR to open the discussion about what makes up a Hub and what things we may need to define within the scope of the spec.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2471,6 +2471,307 @@ <h2>
 
     </section>
 
+  <section class="normative">
+    <h2>Identity Hubs</h2>
```
Comment (OR13) [inline index.html:2475]: may want to add an issue linking to the naming debate issue... for example:

```
<p class="issue" data-number="423">
The Working Group is still debating naming of this instance and its associated interfaces and data model.
</p>
```

File: index.html [hunk 2]
```diff
@@ -2471,6 +2471,307 @@ <h2>
...
+    "data": {...}
+  }
+}
+          </pre>
+
+          Example of an additional decendant modification of a logical object:
+          <pre class="example">
+{ //  Assume the object hash ID is 789
+  "iss": "did:foo:a5b4c3",
+  "@context": "https://identity.foundation/schemas/hub/v0.1",
```
Comment (OR13) [inline index.html:2547]: please open an issue to make this a resolvable URL... https://identity.foundation/schemas/hub/v0.1

File: index.html [hunk 3]
```diff
@@ -2471,6 +2471,307 @@ <h2>
...
+      <h3>Instance Discovery</h3>
+    </section>
+
+    <section>
+      <h3>Object IDs & Relationships</h3>
+
+      <ol>
+        <li>
+          All objects Identity Hubs interact with are identified by the hash of their content. To derive an ID for 
+          an object in the system. To derive the ID of an object, generate a Base64URL-encoded SHA-256 multihash
```
Comment (OR13) [inline index.html:2501]: We need to have a come to jesus moment regarding base58 vs base64url.... can you open an issue regarding this?

File: index.html [hunk 4]
```diff
@@ -2471,6 +2471,307 @@ <h2>
...
+        syncronization, which begins with the DID Document of the controlling entity. The DID Document of a 
+        Hub controller MUST include Service Endpoint entries that allow Hub instances to locate and actively 
+        initiate syncronization amongst themselves.
+
+        These Service Endpoints are defined as follows: ...
+      </p>
+    </section>
+
+    <section>
+      <h3>Object State Syncronization</h3>
```
Comment (OR13) [inline index.html:2595]: glad that we have split replication and synchronization.

File: index.html [hunk 5]
```diff
@@ -2471,6 +2471,307 @@ <h2>
...
+
+          Example of an additional decendant modification of a logical object:
+          <pre class="example">
+{ //  Assume the object hash ID is 789
+  "iss": "did:foo:a5b4c3",
+  "@context": "https://identity.foundation/schemas/hub/v0.1",
+  "@type": "Write",
+  "interface": "Collections",
+  "commit": {
+    "strategy": "crdt-delta",
```
Comment (OR13) [inline index.html:2551]: this should have its own section under synchronization.

**Review by OR13 — CHANGES_REQUESTED** [review]
I think there are 2 sections which are missing from this, which we need to manage alignment.

1. Data Model
1.1 Profiles
1.2 Collections
1.3 Commitments
2. Interfaces 
2.1 CRUD (HTTP + POST)

please add these 2 top level sections under Identity Hubs, and move the examples to these sections, and reference them by link from the associated other sections... 

These 2 sections are where we will need to get crystal clear, and it will be hard chasing examples down if we don't formalize things like this.

File: index.html (hunk 3 above)
Comment (csuwildcat) [reply index.html:2501]: Issue here: https://github.com/decentralized-identity/secure-data-store/issues/119

File: index.html (hunk 5 above)
Comment (csuwildcat) [reply index.html:2551]: I had subsections for different sync strategies, are you talking about something in addition or different?

Comment (OR13) [reply index.html:2551]: then we are good,

**Review by OR13 — APPROVED** [review]
I'd like to merge this, and then do another round of editorial cleanup. we need to get the spec ready to take PRs from the group.

**Review by dmitrizagidulin — APPROVED** [review]
👍

**Comment by OR13** [comment]
@csuwildcat to move appendix A into hubs, and fix the sidebar navigation.

File: index.html [hunk 6]
```diff
@@ -2471,6 +2471,324 @@ <h2>
...
+  </section>
+
+
+  <section>
+    <h4>Collections</h4>
+
+    <p>
+      An entity MAY choose to populate their logical Hub (referring to all instances as a singular mechanism) 
+      with a Profile object that describes the owning entity. The profile object SHOULD use whatever schema(s) 
+      and values that best describe them. The Profile interface is addressed via the following methods:
```
Comment (llorllale) [inline index.html:2726]: copy pasta from above - this paragraph should be about collections

**Review by llorllale — APPROVED** [review]

**Comment by OR13** [comment]
Merging, will need lots of editorial cleanup.

**Pull request merged**